modules/javafx.graphics/src/main/java/com/sun/glass/ui/monocle/EPDFrameBuffer.java

```diff
@@ -617,6 +675,14 @@ long getNativeHandle() {
      * Gets the virtual horizontal resolution of the frame buffer. See the notes
      * for the {@linkplain EPDFrameBuffer#EPDFrameBuffer constructor} above.
      *
+     * @implNote When using an 8-bit frame buffer, set by the system property
+     * {@code monocle.epd.bitsPerPixel=8}, the Kobo Touch Models N905B and N905C
+     * and the Kobo Glo HD Model N437 work only when this method returns
+     * {@code xresVirtual}, while the Kobo Clara HD Model N249 works only when
+     * this method returns {@code xres}.
+     * <p>
+     * TODO: Make 8-bit frame buffers work for all devices.</p>
```

Comment:
One way to work around the bogus frame buffer resolution might be to add a system property that overrides the reported value, such as `monocle.epd.xresVirtual.` Oddly enough, the error does not occur on the Kobo Clara HD when the 8-bit frame buffer is inverted (`-Dmonocle.epd.y8inverted=true`).